mm/src/code/z_bgcheck.c

```diff
@@ -4207,7 +4207,11 @@ WallType SurfaceType_GetWallType(CollisionContext* colCtx, CollisionPoly* poly,
 }
 
 s32 SurfaceType_GetWallFlags(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
-    return sWallFlags[SurfaceType_GetWallType(colCtx, poly, bgId)];
+    s32 result = sWallFlags[SurfaceType_GetWallType(colCtx, poly, bgId)];
+    if (GameInteractor_Should(VB_BE_CLIMBABLE_SURFACE, result & WALL_FLAG_3, poly, bgId)) {
+        result |= WALL_FLAG_3;
+    }
+    return result;
```

Comment:
Instead of `result & WALL_FLAG_3` the vanilla condition should be `false` because the behavior within the should never happens in vanilla. 
Additionally, neither `poly` nor `bgId` are used in the hook, and should be removed here. 
I'd also like the original return line to remain unchanged in the vanilla case.

```suggestion
    if (GameInteractor_Should(VB_BE_CLIMBABLE_SURFACE, false)) {
        return sWallFlags[SurfaceType_GetWallType(colCtx, poly, bgId)] | WALL_FLAG_3;
    }
    return sWallFlags[SurfaceType_GetWallType(colCtx, poly, bgId)];
```

Reply:
Went ahead and committed your exact suggested change—early return for the cheat, vanilla return untouched otherwise. I lean DRY out of habit, but I get the case for vanilla return flow in a project like this.

Thanks for the feedback!